vendor/parser-extended/parser.mly

```diff
@@ -189,6 +189,13 @@ let mkuplus ~oploc name arg =
   match name, desc with
   | "+", Pexp_constant({pconst_desc= Pconst_integer _; _})
   | ("+" | "+."), Pexp_constant({pconst_desc= Pconst_float _; _}) -> desc
+
+  (* Jane Street extension *)
+  | "+", Pexp_constant({pconst_desc= Pconst_unboxed_integer _; _})
+  | ("+" | "+."), Pexp_constant({pconst_desc= Pconst_unboxed_float _; _})
```

Comment:
I have the same question here I had before:  Probably we don't want the "+." syntax since we don't otherwise have the dotted arithmetic operators for float#?

Reply:
The weird thing is that this `+.` isn't really the `+.` operator. It's just a bit of concrete syntax. That is, even if you redefine `+.` to mean something else, we accept `+.3.0` as identical to `3.0`. I find this all a bit bizarre. But it's what we have. So I decided to just treat unboxed floats the same as boxed ones in this regard.

On the other hand, if the argument of the `+.` is not a literal, then it uses the prefix `(~+.)` operator that's in scope.

Making this concrete, take a look at this:

```ocaml
let (~+.) x = x *. x
let a = +. 3.0
let b = let x = 3.0 in +. x
```

This evaluates to have `a = 3.0` but `b = 9.0`.

So I maintain that we should just propagate the current behavior.

Reply:
I'm not sure I understand your argument.

You and I may understand that `+.` is a weird special piece of syntax in front of a literal, having looked at this, but I don't expect most programmers do.  So, the fact that we'll accept `+. #3.14` but not `#3.14 +. #3.14` seems like a confusing fact we're destined to get complaints about.  That is, I'd say that to whatever limited extent the current behavior makes sense for floats, it's only because of the existence of `+.` and `~+.`, and we don't have those for unboxed floats.

Obvious this is a small thing, so if that didn't convince you, go ahead and resolve this and merge.

Reply:
But right now OCaml accepts `+ 3.0` but not `let x = 3.0 in + x`. Have we gotten complaints?

Reply:
It sounds like I haven't convinced you.  That's fine - go ahead and merge.